import/src/main/kotlin/com/ritense/import/ValtimoImportService.kt

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright 2015-2023 Ritense BV, the Netherlands.
+ *
+ * Licensed under EUPL, Version 1.2 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" basis,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ritense.import
+
+import java.io.InputStream
+import java.util.zip.ZipInputStream
+import mu.KLogger
+import mu.KotlinLogging
+import org.springframework.transaction.annotation.Transactional
+
+open class ValtimoImportService(
+    importers: Set<Importer>
+) : ImportService {
+
+    private val orderedImporters: LinkedHashSet<Importer>
+
+    init {
+        //Check for duplicate import types in the list of importers
+        val wImporters = importers.map { WrappedImporter(it) }.toSet()
+        if (wImporters.size != importers.size) {
+            val duplicatedTypes = wImporters.filter { wImporter ->
+                importers.count { wImporter.type() == it.type() } > 1
+            }.map { it.type() }
+            throw ImportServiceException(
+                "Multiple importers of the same type provided: [${duplicatedTypes.joinToString()}]"
```

Comment:
The main goal for the typed exception was to be able to throw a status 400 instead of a 500. Because this will be thrown at startup, that scenario doesn't even apply.

But while in general you are right, I don't like having I don't know how many specific Exception classes that we handle the same regardless.